### PR TITLE
Add mermaid diagram support

### PR DESCRIPTION
**Pull request by dneto0**
Use it in a code block like this

```mermaid
flowchart TD
  Start --> Stop
```

### CODE REVIEW

**Comment by dneto0** [comment]
Ha, that should read:
```
    ```mermaid
    flowchart TD
          Start --> Stop
    ```
    
```

File: layouts/partials/mermaid.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{{ if .Page.Store.Get "hasMermaid" }}
+  <script type="module">
+    import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.esm.min.mjs';
```
Comment (dj2) [inline layouts/partials/mermaid.html:3]: Do we want to be pulling this from a CDN or should we make a local copy of the `mermaid.esm.mjs` file?

File: .github/workflows/hugo.yml [hunk 2]
```diff
@@ -48,6 +48,8 @@ jobs:
         uses: actions/configure-pages@v3
       - name: Install Node.js dependencies
         run: "[[ -f package-lock.json || -f npm-shrinkwrap.json ]] && npm ci || true"
+      - name: Bundle Mermaid
+        run: "[[ -f assets/third_party/mermaid/dist ]] || npm run build"
```
Comment (dneto0) [inline .github/workflows/hugo.yml:52]: Haven't tried this yet...

**Comment by dj2** [comment]
As a simpler alternative, we can set this up using esbuild and export `Mermaid` into the window which lets it get picked up by the script later. Just have to make sure the `app.ts` script is loaded before mermaid is initialized.

I pulled down this PR and made the updates, you can see the diff at:

https://github.com/google/tour-of-wgsl/compare/main...dj2:tour-of-wgsl:add-mermaid-support

**Comment by dj2** [comment]
Adding Ben as a reviewer as I've done some of the stuff in here now.

**Pull request merged**